hacep/src/main/java/org/kie/hacep/core/KieSessionContext.java

```diff
@@ -15,31 +15,49 @@
  */
 package org.kie.hacep.core;
 
+import java.util.concurrent.TimeUnit;
+
 import org.kie.api.runtime.KieSession;
+import org.kie.api.time.SessionClock;
+import org.kie.api.time.SessionPseudoClock;
 import org.kie.hacep.consumer.FactHandlesManager;
 import org.kie.hacep.core.infra.SnapshotInfos;
 
 public class KieSessionContext {
 
     private KieSession kieSession;
 
+    private SessionPseudoClock clock;
+
     private FactHandlesManager fhManager;
 
     public KieSession getKieSession() {
         return kieSession;
     }
 
     public void initFromSnapshot(SnapshotInfos infos) {
-        this.kieSession = infos.getKieSession();
+        setKieSession(infos.getKieSession());
         this.fhManager = infos.getFhManager();
     }
 
     public void init(KieSession newKiession) {
-        this.kieSession = newKiession;
+        setKieSession(newKiession);
         this.fhManager = new FactHandlesManager(newKiession);
     }
 
+    private void setKieSession(KieSession kieSession) {
+        this.kieSession = kieSession;
+        SessionClock clock = kieSession.getSessionClock();
+        if (clock instanceof SessionPseudoClock) {
+            this.clock = (SessionPseudoClock) clock;
+        }
```

Comment:
There is no check that clock type is `pseudo`, this should fail if in streaming mode or at least add a warning/info message. Wdyt?

Reply:
Not understanding this remark: the check is the instanceof or am I missing something?

Reply:
My only point is that if the session is not properly configured, there is no error/message. It will skip this statement and then it will fail as soon as it will perform the first insert

Reply:
At that point we don't know if the ksession is a cep one or not, and in this second case we don't mind about the clock. I'll add an error log line in the setClockAt method.